ddl/index.go

```diff
@@ -31,14 +31,20 @@ import (
 	"github.com/pingcap/tidb/terror"
 	"github.com/pingcap/tidb/util"
 	"github.com/pingcap/tidb/util/types"
+	"math"
 )
 
-const maxPrefixLength = 767
+const maxPrefixLength = 3072
 
-func buildIndexInfo(tblInfo *model.TableInfo, unique bool, indexName model.CIStr,
-	idxColNames []*ast.IndexColName) (*model.IndexInfo, error) {
+// BuildIndexInfo validates index length and create index info
+func BuildIndexInfo(tblInfo *model.TableInfo, indexName model.CIStr,
```

Comment:
There are too many arguments. Can an initial `*model.IndexInfo` be used as an incoming argument?